tests/model_forms/test_modelchoicefield.py

```diff
@@ -190,6 +204,12 @@ def test_disabled_modelchoicefield_has_changed(self):
         field = forms.ModelChoiceField(Author.objects.all(), disabled=True)
         self.assertIs(field.has_changed('x', 'y'), False)
 
+    def test_disabled_and_instance_initial_modelchoicefield(self):
+        class TestForm(forms.Form):
+            categories = forms.ModelChoiceField(Category.objects.all(), disabled=True, initial=self.c1)
+
+        self.assertTrue(TestForm(data={"categories": self.c1.pk}).is_valid())
```

Comment:
Can we just do this with the field, calling `clean()` and having it **not** raise the `ValidationError`?

Reply:
May we remove the entire test `test_disabled_and_instance_initial_modelchoicefield` as `clean()` is checked in your assertion `self.assertEqual(f.clean(self.c1), self.c1)`.

Reply:
Right, but it's the `disabled` bit that breaks right? (Wasn't the weirdness from the ticket that exactly the same thing worked or not depending on that value? So a regression test for that is worth having.)

Reply:
**disabled bit active :** the `initial` value is cleaned
**disabled bit not active :** the `POST` value is cleaned

If we replace this test by a `clean(Field.initial)`, it is the same you added in basics test. If we wan't to reproduce the ticket, we can keep the full test with the `Form.is_valid` wich select `inital` or `POST` depending on disabled bit.

Reply:
Fine. Super. Thanks for the clarification. (In that case, leave it as it is, because we want the test for the issue...)